### PR TITLE
Fixes #42: Don't use diff-so-fancy for core pager

### LINKED ISSUE #42: Current git config instructions clobbers `git log`
`git config --global core.pager "diff-so-fancy | less --tabs=1,5 -R"`

This highjacks things like `git log`, and other commands locally for me. Is there a non-destructive way of doing this? If not, I'd suggest showing how to alias it in bash instead of adding it to the gitconfig.

### PR DESCRIPTION
**Pull request by noscript**
Addresses #42.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -21,7 +21,8 @@ git diff --color | diff-so-fancy
 
 **But**, you'll probably want to fancify all your diffs. Run this so `git diff` will use it:
 ```shell
-git config --global core.pager "diff-so-fancy | less --tabs=1,5 -R"
+git config --global pager.diff "diff-so-fancy | less --tabs=1,5 -R"
```
Comment (stevemao) [inline readme.md]: I'd say something like "If you only want to use it with git-diff, run blabla".

Comment (stevemao) [inline readme.md]: Oh I see, this is trying to avoid git-log using the same pager.

**Comment by stevemao** [comment]
LGTM, is there any other command that might work great with our script?

**Comment by noscript** [comment]
`git add --patch` produces diff output but I'm not sure how to set a pager for that.

File: readme.md [hunk 2]
```diff
@@ -71,7 +72,7 @@ You may also want to configure [general diff colors](https://github.com/pauliris
 Sometimes you will want to bypass diff-so-fancy. Easy enough:
 
 ```shell
-git --no-pager diff  # will avoid core.pager hook
+git --no-pager diff  # will avoid pager.diff hook
```
Comment (stevemao) [inline readme.md]: I think this should say "will avoid any pager hook"

File: readme.md [hunk 3]
```diff
@@ -68,10 +69,10 @@ You may also want to configure [general diff colors](https://github.com/pauliris
 
 ## Opting-out
 
-Sometimes you will want to bypass diff-so-fancy. Easy enough:
+Sometimes you will want to bypass diff-so-fancy. Use `--no-pager` for that:
```
Comment (stevemao) [inline readme.md]: I prefer the original wording but I'll let @paulirish to make the call :)

**Comment by paulirish** [comment]
Yup. This is the best way to handle it.

**Comment by vassudanagunta** [comment]
> `git add --patch` produces diff output but I'm not sure how to set a pager for that. #35

Not possible I think. See my answer to [How do I use a pager for long git add --patch hunks?](https://stackoverflow.com/questions/39743490/how-do-i-use-a-pager-for-long-git-add-patch-hunks/63652969#63652969). Does anyone disagree?

**Pull request merged**